src/plugins/analysis/hash/code/hash.py

```diff
@@ -25,7 +25,7 @@ def __init__(self, plugin_administrator, config=None, recursive=True):
 
         # additional init stuff can go here
 
-        super().__init__(plugin_administrator, config=config, recursive=recursive, plugin_path=__file__)
+        super().__init__(plugin_administrator, config=config, recursive=recursive, plugin_path=__file__, timeout=600)
```

Comment:
maybe we should split the plugin in the long run into a "basic hash" plugin for mandatory hashes and an "advanced hash" plugin for hashes that could take a while or even fail to compute

Reply:
I agree. But since tlsh is a dependecy for compare we then either make the advanced hashes mandatory as well or we have to account for that in the compare. This is also a lead to cross-plugin results since tlsh is calculated in the tlsh plugin (which depends, like compare, on tlsh being present for all files (although in this case the non existence is covered) and imphash can be easily calculated using the results of elf_analysis